### ISSUE TITLE
Value not found at "self" source

### ISSUE THREAD

**Issue by FelCer** [serverless/examples#689]
Hello good day, I am trying to implement 'aws-node-rest-api-with-dynamodb-and-offline' but when I run the command `serverless dynamodb install` it gives me the following errors:
```
Error:
Cannot resolve serverless.yml: Variables resolution errored with:
  - Cannot resolve variable at "provider.environment.DYNAMODB_TABLE": Value not found at "self" source,
  - Cannot resolve variable at "provider.iamRoleStatements.0.Resource": Value not found at "self" source
```
And I am executing as indicated in the guide(https://www.serverless.com/examples/aws-node-rest-api-with-dynamodb-and-offline, https://github.com/serverless/examples/tree/master/aws-node-rest-api-with-dynamodb-and-offline)

![image](https://user-images.githubusercontent.com/22477264/159971489-41da1ab3-b6eb-4cd7-8f29-713c95b06623.png)
![image](https://user-images.githubusercontent.com/22477264/159971597-cc2c68b5-26dc-4c38-8530-74fa47e0c0a3.png)

**Comment by FelCer** [comment]
That's fine if it worked, but I _think_ this template still contains errors, like:
executing the `serverless offline start` command produces the following error.
![image](https://user-images.githubusercontent.com/22477264/160126871-d39ae49e-32a7-4555-a947-9dd3e2a12417.png)

This error is treated here, https://github.com/99x/serverless-dynamodb-local/issues/194 but without any solution.
By implementing that with the flag indicated here https://github.com/99x/serverless-dynamodb-local/issues/194#issuecomment-590883319 it runs correctly, but until the endpoint is consulted it generates an error.
![image](https://user-images.githubusercontent.com/22477264/160127212-b9a3b6c8-1181-4898-9e6d-e06d539977de.png)
![image](https://user-images.githubusercontent.com/22477264/160127314-28471ba2-12d8-4440-8708-a1113e421695.png)

**Comment by pgrzesik** [comment]
Hey @FelCer - it's very likely that this example might have additional errors, since it hasn't been actively maintained for a long time - unfortunatelly I don't know much about this specific example so I won't be able to help more there

**Comment by gsanjeev432** [comment]
Actually, there is no value to satisfy the variable "stage" and "region". So you need to either pass it from command line as `serverless` deploy --stage dev --region us-east-1` or else define the variables in serverless.yml.

**Comment by eriklive** [comment]
> Yeah, maybe I'm a little too late, but appears to me that in "Provider" section should be a "region" element, dont?

That's right, I was facing the same issue. Here is an example that servers both ways:

![image](https://user-images.githubusercontent.com/12226731/205930229-ebbd65c4-fe6c-4c58-8c84-6f6b4c0bc2d7.png)


The key is `${self:custom.integrarProdutoMLFunctionSQS.${opt:stage}}`  on lines 12 and 16. 

The `self:custom.integrarProdutoMLFunctionSQS` will refeer to **custom** section, **integrarProdutoMLFunctionSQS** subsection. 

But I need thoose variables being environment dependant, so I run my  serverless by `serverless offline start --stage local`. To access my "stage" variable, set throught console, I need to use `${opt:stage}`. 

In conclusion:

1. To use file values, use `${self:section.value}` (like me, using  `self:custom.integrarProdutoMLFunctionSQS`)
2. To use varibles set throught console, use `${opt:stage}`

So when I do use  `${self:custom.integrarProdutoMLFunctionSQS.${opt:stage}}`, running with `serverless offline start --stage local`, the final result will be `${self:custom.integrarProdutoMLFunctionSQS.local}`, wich is what I need.

**Comment by ChuhengH** [comment]
is the ${self:someProperty} used in the rate(Value Unit), such as rate(${self:someProperty})?
I found it doesn't work.
But rate(${env:SOME_VAR}) will work. Why?
